sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py

```diff
@@ -320,3 +562,65 @@ def test_get_trial_metrics(katib_client, test_name, kwargs, expected_output):
     except Exception as e:
         assert type(e) is expected_output
     print("test execution complete")
+
+
+@pytest.mark.parametrize("test_name,kwargs,expected_output", test_tune_data)
+def test_tune(katib_client, test_name, kwargs, expected_output):
+    """
+    test tune function of katib client
+    """
+    print("\n\nExecuting test:", test_name)
+
+    PYTORCHJOB_KIND = "PyTorchJob"
+    JOB_PARAMETERS = {
+        "PyTorchJob": {
+            "model": "KubeflowOrgV1PyTorchJob",
+            "plural": "pytorchjobs",
+            "container": "pytorch",
+            "base_image": "docker.io/pytorch/pytorch:2.1.2-cuda11.8-cudnn8-runtime",
+        }
+    }
+
+    with patch.dict(
+        "sys.modules",
+        {
+            "kubeflow.storage_initializer": Mock(),
+            "kubeflow.storage_initializer.hugging_face": Mock(),
+            "kubeflow.storage_initializer.s3": Mock(),
+            "kubeflow.storage_initializer.constants": Mock(),
+            "kubeflow.training": MagicMock(),
+            "kubeflow.training.models": Mock(),
+            "kubeflow.training.utils": Mock(),
+            "kubeflow.training.constants": Mock(),
+            "kubeflow.training.constants.constants": Mock(),
+        },
+    ), patch(
+        "kubeflow.storage_initializer.hugging_face.HuggingFaceModelParams",
+        HuggingFaceModelParams,
+    ), patch(
+        "kubeflow.storage_initializer.hugging_face.HuggingFaceDatasetParams",
+        HuggingFaceDatasetParams,
+    ), patch(
+        "kubeflow.storage_initializer.hugging_face.HuggingFaceTrainerParams",
+        HuggingFaceTrainerParams,
+    ), patch(
+        "kubeflow.storage_initializer.s3.S3DatasetParams", S3DatasetParams
+    ), patch(
+        "kubeflow.training.models.KubeflowOrgV1PyTorchJob", KubeflowOrgV1PyTorchJob
+    ), patch(
+        "kubeflow.training.constants.constants.JOB_PARAMETERS", JOB_PARAMETERS
+    ), patch(
+        "kubeflow.training.constants.constants.PYTORCHJOB_KIND", PYTORCHJOB_KIND
+    ), patch(
+        "kubeflow.katib.utils.utils.get_trial_substitutions_from_trainer",
+        return_value={"param": "value"},
+    ), patch.object(
+        katib_client, "create_experiment", return_value=Mock()
+    ) as mock_create_experiment:
+        try:
+            katib_client.tune(**kwargs)
+            mock_create_experiment.assert_called_once()
+            assert expected_output == TEST_RESULT_SUCCESS
```

Comment:
Is there a way for you to validate which Experiment object we will create here: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L715 ?
I think, that will allow you to make sure we create the correct Experiment object for various test cases.

Reply:
@andreyvelich I previously added a similar check in the following section of the code (which is deleted now):
https://github.com/kubeflow/katib/blob/e4f7922ac448257485dcd6377d2b4aa663ba0d15/test/unit/v1beta1/tune-api/test_tune_api.py#L487-L887
One concern I have is that when users want to optimize hyperparameters in LLMs, the container arguments will include the JSON transformation of `training_parameters` (from the `transformers.TrainingArguments` class) and `lora_config` (from the `peft.LoraConfig` class). Since these classes are defined outside our scope, any future changes to them could cause the actual results in the unit tests to no longer match the expected results, leading to test failures. I'm not sure if this concern is valid. Do you have any suggestions?

Reply:
Maybe you don't need to validate the all arguments ? Just make sure that arguments contain the hyperparameters that we want to optimize. E.g. `"${trialParameters.r}"`

Reply:
OK got it! I will add that.